### PR TITLE
fix(@desktop/wallet): [SwapModal] Fix for flaky tests and remove the skip flag

### LINKED ISSUE #15272: [SwapModal] Fix flaky tests and remove the `skip` flag
#### Description

The purpose of this task is to fix `test_modal_swap_proposal_setup` and `test_floating_header_list_items` and remove the skip flag.

It seems like these tests depend on some initialisation order because it's randomly failing.

              @Khushboo-dev-cpp I'll skip 2 tests here because it's randomly failing on master. Do you want me to add a task for them, or should I mention them in an existing one?

_Originally posted by @alexjba in https://github.com/status-im/status-desktop/pull/15099#discussion_r1647050544_

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixes #15272

### What does the PR do

Fixes flaky tests

### Affected areas

tst_SwapModal.qml

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

### CODE REVIEW

**Comment by Khushboo-dev-cpp** [comment]
@alexjba or @caybro  if you kindly test this once on your system and let me know if the tests run consistently for you? Ive tried running them back to back 20 times and always passed.
I wasnt able to reproduce the issue even before the tests were commented out so its best if one of you who faced it too can check :)

File: storybook/qmlTests/tests/tst_SwapModal.qml [hunk 1]
```diff
@@ -56,18 +56,13 @@ Item {
     readonly property var tokenSelectorAdaptor: TokenSelectorViewAdaptor {
         assetsModel: swapAdaptor.walletAssetsStore.groupedAccountAssetsModel
         flatNetworksModel: swapStore.flatNetworks
-        currentCurrency: swapAdaptor.currencyStore
+        currentCurrency: swapAdaptor.currencyStore.currentCurrency
```
Comment (caybro) [inline storybook/qmlTests/tests/tst_SwapModal.qml:59]: Ouch, my bad

**Review by caybro — APPROVED** [review]
LGTM!

Pending the tests passing ofc :wink:

**Comment by caybro** [comment]
This is a new failure though:
```
[2024-06-25T21:48:38.515Z] 3: FAIL!  : QmlTests::SwapInputPanel::test_input_greater_than_max_balance() Uncaught exception: Cannot read property 'marketDetails' of null
[2024-06-25T21:48:38.515Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15343/storybook/qmlTests/tests/tst_SwapInputPanel.qml(405)]
```

**Comment by Khushboo-dev-cpp** [comment]
> This is a new failure though:
> 
> ```
> [2024-06-25T21:48:38.515Z] 3: FAIL!  : QmlTests::SwapInputPanel::test_input_greater_than_max_balance() Uncaught exception: Cannot read property 'marketDetails' of null
> [2024-06-25T21:48:38.515Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15343/storybook/qmlTests/tests/tst_SwapInputPanel.qml(405)]
> ```

oops seems like SwapInputPanel tests are failing. Will try fixing it, might be a flaky one

**Comment by caybro** [comment]
> > This is a new failure though:
> > ```
> > [2024-06-25T21:48:38.515Z] 3: FAIL!  : QmlTests::SwapInputPanel::test_input_greater_than_max_balance() Uncaught exception: Cannot read property 'marketDetails' of null
> > [2024-06-25T21:48:38.515Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15343/storybook/qmlTests/tests/tst_SwapInputPanel.qml(405)]
> > ```
> 
> oops seems like SwapInputPanel tests are failing. Will try fixing it, might be a flaky one

Hmm, I can't reproduce locally and never seen it fail in a PR before

**Comment by Khushboo-dev-cpp** [comment]
@alexjba I tried to setup latest qt@5 (Qt5.15.13) on my intel mac to find out why these tests are failing for you. But I have na issues running the tests themselves! I request you to please check this as I am unable to reproduce this issue!

**Comment by alexjba** [comment]
> @alexjba I tried to setup latest qt@5 (Qt5.15.13) on my intel mac to find out why these tests are failing for you. But I have na issues running the tests themselves! I request you to please check this as I am unable to reproduce this issue!

@Khushboo-dev-cpp I'll try a fix later this week. Is this ok with you, or you think we should fix it earlier?

**Comment by Khushboo-dev-cpp** [comment]
@alexjba the earlier the better so that future PRs don't break, however as it is not reproducible for Lukas on mac os with Qt5.15.8. Ive also asked @dlipicar to check once on his mac os and if its not reproducible for him as well, we can merge this PR and keep the issue open and assigned to you, and you can get to it when you have the time.

**Comment by alexjba** [comment]
> @alexjba the earlier the better so that future PRs don't break, however as it is not reproducible for Lukas on mac os with Qt5.15.8. Ive also asked @dlipicar to check once on his mac os and if its not reproducible for him as well, we can merge this PR and keep the issue open and assigned to you, and you can get to it when you have the time.

Sure, if the CI is stable! 🎉

**Review by alexjba — APPROVED** [review]

**Pull request merged**